src/secrets/CommandEdit.ts

```diff
@@ -54,41 +54,66 @@ class CommandEdit extends CommandPolykey {
           },
           logger: this.logger.getChild(PolykeyClient.name),
         });
-        const response = await binUtils.retryAuthentication(
-          (auth) =>
-            pkClient.rpcClient.methods.vaultsSecretsGet({
-              metadata: auth,
-              nameOrId: secretPath[0],
-              secretName: secretPath[1],
-            }),
-          meta,
-        );
-        const secretContent = response.secretContent;
+        const [secretExists, secretContent] =
+          await binUtils.retryAuthentication(async (auth) => {
+            let secretExists: boolean = true;
+            let response;
+            try {
+              response = await pkClient.rpcClient.methods.vaultsSecretsGet({
+                metadata: auth,
+                nameOrId: secretPath[0],
+                secretName: secretPath[1],
+              });
+            } catch (e) {
+              if (e.cause.name === 'ErrorSecretsSecretUndefined') {
+                secretExists = false;
+              } else {
+                throw e;
+              }
```

Comment:
This is not how we check the type of secret. We want to use `instanceof` here.

```ts
if (e.cause instanceof ErrorSecretsSecretUndefined) ...
```

I also think we have a utility for unwrapping remote errors. It's called `remoteErrorCause` in the utilities. This will return the cause for that error.

On top of that you need to use `instanceof` to check its actually an instance of that error.